test/ResourceCacheSpec.js

```diff
@@ -88,16 +88,22 @@ describe('ResourceCache', function() {
   });
 
   describe('#fetchAllDirty', function() {
-    it('returns all dirty records in the table', function(done) {
+    function getCache() {
       var cache = Object.create(ResourceCache);
       cache.setSchemaBuilder(getSchemaBuilder());
       cache.setTableName('my_table');
       cache.setStoreType(lf.schema.DataStoreType.MEMORY);
       cache.createTable();
+
+      return cache;
+    }
+
+    it('returns all dirty records in the table', function(done) {
+      var cache = getCache();
       cache.persist({}).then(function(records) {
         var uuid = records[0].uuid;
         cache.fetchAllDirty().then(function(records) {
-          if (records[0].uuid === uuid) {
+          if (records.length === 1 && records[0].uuid === uuid) {
```

Comment:
This may be a silly question, but where is the 1 record coming from?  I looked in the file and in `cbit.ResourceCache` to no avail.